src/fixup/fixup_radc2p.cpp

```diff
@@ -0,0 +1,364 @@
+            fail(k, j + 1, i) > 0.5
+                ? v(b, iv, k, j, i) = std::min(std::fabs(v(b, iv, k, j, i)),
+                                               std::fabs(v(b, iv, k, j + 1, i)))
+                : v(b, iv, k, j, i);
+            if (ndim > 2) {
+              fail(k - 1, j, i) > 0.5
+                  ? v(b, iv, k, j, i) = std::min(std::fabs(v(b, iv, k, j, i)),
+                                                 std::fabs(v(b, iv, k - 1, j, i)))
+                  : v(b, iv, k, j, i);
+              fail(k + 1, j, i) > 0.5
+                  ? v(b, iv, k, j, i) = std::min(std::fabs(v(b, iv, k, j, i)),
+                                                 std::fabs(v(b, iv, k + 1, j, i)))
+                  : v(b, iv, k, j, i);
+            }
+          }
+        };
+
+        if (v(b, iradfail, k, j, i) == radiation::FailFlags::fail) {
+          const Real sdetgam = geom.DetGamma(CellLocation::Cent, k, j, i);
+          Real gcov[4][4];
+          geom.SpacetimeMetric(CellLocation::Cent, k, j, i, gcov);
+          const Real vel[] = {v(b, idx_pvel(0), k, j, i), v(b, idx_pvel(1), k, j, i),
+                              v(b, idx_pvel(2), k, j, i)};
+          const Real W = phoebus::GetLorentzFactor(vel, gcov);
+          Vec con_v({vel[0] / W, vel[1] / W, vel[2] / W});
+
+          typename CLOSURE::LocalGeometryType g(geom, CellLocation::Cent, b, k, j, i);
+
+          Real num_valid = v(b, iradfail, k, j, i - 1) + v(b, iradfail, k, j, i + 1);
+          if (ndim > 1)
+            num_valid += v(b, iradfail, k, j - 1, i) + v(b, iradfail, k, j + 1, i);
+          if (ndim == 3)
+            num_valid += v(b, iradfail, k - 1, j, i) + v(b, iradfail, k + 1, j, i);
+          if (rad_c2p_failure_strategy == FAILURE_STRATEGY::interpolate_previous) {
+            for (int ispec = 0; ispec < nspec; ispec++) {
+              v(b, idx_J(ispec), k, j, i) = fixup0(idx_J(ispec));
+              SPACELOOP(ii) {
+                v(b, idx_H(ispec, ii), k, j, i) = fixup0(idx_H(ispec, ii));
+              }
+            }
+          } else if (num_valid > 0.5 &&
+                     rad_c2p_failure_strategy == FAILURE_STRATEGY::neighbor_minimum) {
+            for (int ispec = 0; ispec < nspec; ispec++) {
+              get_minimum(idx_J(ispec));
+              SPACELOOP(ii) { get_minimum(idx_H(ispec, ii)); }
+            }
+          } else {
+            if (num_valid > 0.5 &&
+                rad_c2p_failure_strategy == FAILURE_STRATEGY::interpolate) {
+              const Real norm = 1.0 / num_valid;
+              for (int ispec = 0; ispec < nspec; ispec++) {
+                v(b, idx_J(ispec), k, j, i) = fixup(idx_J(ispec), norm);
+                SPACELOOP(ii) {
+                  v(b, idx_H(ispec, ii), k, j, i) = fixup(idx_H(ispec, ii), norm);
+                }
+              }
+            } else {
+              for (int ispec = 0; ispec < nspec; ispec++) {
+                v(b, idx_J(ispec), k, j, i) = 1.e-100;
```

Comment:
I've had some issues doing that with e.g. checking that variables are greater than `robust::SMALL()` as DBCs. I'm not sure what the right pattern is since this right here is a magic number.

Reply:
`10*robust::SMALL()`?

Reply:
Done :+1: